src/Cli/Microsoft.DotNet.Cli.Utils/Microsoft.DotNet.Cli.Utils.csproj

```
@@ -23,10 +23,10 @@
   <Target Name="VerifyMSBuildDependency" BeforeTargets="ResolveAssemblyReferences" Condition="'$([MSBuild]::GetTargetFrameworkIdentifier($(TargetFramework)))' == '.NETCoreApp'">
     <!-- We explicitly reference an older version of MSBuild here to support VS
     for Mac and other VS scenarios. During source-build, we only have access to
-    the latest version, which targets net7.0. -->
+    the latest version, which targets NetCurrent. -->
     <PropertyGroup>
       <MSBuildPathInPackage>$(PkgMicrosoft_Build_Runtime)\contentFiles\any\net7.0\MSBuild.dll</MSBuildPathInPackage>
-      <MSBuildPathInPackage Condition="'$(DotNetBuildFromSource)' == 'true'">$(PkgMicrosoft_Build_Runtime)\contentFiles\any\net7.0\MSBuild.dll</MSBuildPathInPackage>
+      <MSBuildPathInPackage Condition="'$(DotNetBuildFromSource)' == 'true'">$(PkgMicrosoft_Build_Runtime)\contentFiles\any\$(NetCurrent)\MSBuild.dll</MSBuildPathInPackage>
```

We could probably do this to support both repo-local sourcebuild and VMR sourcebuild in the transition time

```suggestion
      <MSBuildPathInPackage Condition="'$(DotNetBuildFromSource)' == 'true' and !Exists($(MSBuildPathInPackage))">$(PkgMicrosoft_Build_Runtime)\contentFiles\any\$(NetCurrent)\MSBuild.dll</MSBuildPathInPackage>
```

---

This is because the repo level msbuild source-build which produces the intermediate consumed by the SDK's repo level source-build, i[s targeting .NET7.0](https://github.com/dotnet/msbuild/blob/main/Directory.Build.props#L32).  In the full product source-build this targets .net8.0 hence this line needs updating.  Your proposed change with the `and` would work.